src/api/Elastic.Documentation.Api.Core/Search/SearchUsecase.cs

```diff
@@ -0,0 +1,50 @@
+// Licensed to Elasticsearch B.V under one or more agreements.
+// Elasticsearch B.V licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information
+
+using Elastic.Documentation.Api.Core.Validation;
+
+namespace Elastic.Documentation.Api.Core.Search;
+
+public class SearchUsecase(ISearchGateway searchGateway)
+{
+	public async Task<SearchResponse> Search(SearchRequest request, Cancel ctx = default)
+	{
+		// var validationResult = validator.Validate(request);
```

Comment:
This is going to be nicer with .NET 10 for minimal API's.

https://medium.com/@denmaklucky/validation-in-minimal-api-whats-new-in-net-10-9ab6896e9852